pkg/controller/schedulingmanager/controller.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedulingmanager
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	corev1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
+	fedclientset "github.com/kubernetes-sigs/federation-v2/pkg/client/clientset/versioned"
+	corev1alpha1client "github.com/kubernetes-sigs/federation-v2/pkg/client/clientset/versioned/typed/core/v1alpha1"
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/schedulingpreference"
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+	"github.com/kubernetes-sigs/federation-v2/pkg/schedulingtypes"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+var SchedulingRegistry = map[string][]string{
```

Comment:
Would it ever be desirable for a federated type to have more than one scheduler targeting it?  If not, this mapping should probably be template kind to scheduling kind.

cc: @irfanurrehman

Reply:
Apologies, missed this particular comment earlier.
As far as I know and can think, I can't see a scenario where a single federated type is controlled by more then 1 scheduler at the same time. It would obviously result in scheduling conflicts. I can think of a scenario where users don't like the high level scheduler that's packed with open source release and they write their custom scheduler to replace it.
On the other hand a single scheduler can target and/or manipulate multiple federated types (as is the case right now for FederatedDeployments and FederatedReplicaSets being targeted by FederatedRSP).

Reply:
@irfanurrehman Thanks!

@xunpan As per this exchange, please switch to mapping federated type to the appropriate scheduler.  This should allow scheduler start to be simplified.